spring-web/src/test/java/org/springframework/http/server/reactive/MultipartIntegrationTests.java

```diff
@@ -0,0 +1,164 @@
+		return handler;
+	}
+
+	@Test
+	public void getParts() throws Exception {
+		RestTemplate restTemplate = new RestTemplate();
+		RequestEntity<MultiValueMap<String, Object>> request = RequestEntity
+				.post(new URI("http://localhost:" + port + "/parts"))
+				.contentType(MediaType.MULTIPART_FORM_DATA)
+				.body(generateBody());
+		ResponseEntity<Void> response = restTemplate.exchange(request, Void.class);
+		assertEquals(HttpStatus.OK, response.getStatusCode());
+	}
+
+	@Test
+	public void getAllParts() throws Exception {
+		RestTemplate restTemplate = new RestTemplate();
+		RequestEntity<MultiValueMap<String, Object>> request = RequestEntity
+				.post(new URI("http://localhost:" + port + "/all-parts"))
+				.contentType(MediaType.MULTIPART_FORM_DATA)
+				.body(generateBody());
+		ResponseEntity<Void> response = restTemplate.exchange(request, Void.class);
+		assertEquals(HttpStatus.OK, response.getStatusCode());
+	}
+
+	private MultiValueMap<String, Object> generateBody() {
+		HttpHeaders fooHeaders = new HttpHeaders();
+		fooHeaders.setContentType(MediaType.TEXT_PLAIN);
+		ClassPathResource fooResource = new ClassPathResource("org/springframework/web/multipart/reactive/foo.txt");
+		HttpEntity<ClassPathResource> fooPart = new HttpEntity<>(fooResource, fooHeaders);
+		HttpEntity<String> barPart = new HttpEntity<>("bar");
+		MultiValueMap<String, Object> parts = new LinkedMultiValueMap<>();
+		parts.add("fooPart", fooPart);
+		parts.add("barPart", barPart);
+		return parts;
+	}
+
+	public static class CheckRequestHandler implements WebHandler {
+
+		@Override
+		public Mono<Void> handle(ServerWebExchange exchange) {
+
+			if (exchange.getRequest().getURI().getPath().equals("/parts")) {
+				return assertGetParts(exchange);
+			}
+			else if (exchange.getRequest().getURI().getPath().equals("/all-parts")) {
+				return assertGetAllParts(exchange);
+			}
+			return Mono.error(new AssertionError());
+		}
+
+		private Mono<Void> assertGetParts(ServerWebExchange exchange) {
+			return exchange.getParts()
+					.log("reactor", Level.SEVERE)
+					.doOnNext(part -> {
+						if (part.getName().equals("fooPart")) {
+							assertEquals("fooPart", part.getName());
+							Optional<String> filename = part.getFilename();
+							assertTrue(filename.isPresent());
+							assertEquals("foo.txt", filename.get());
+							DataBuffer buffer = part
+									.getContent()
+									.reduce((s1, s2) -> s1.write(s2))
+									.block();
+							assertEquals(12, buffer.readableByteCount());
+							byte[] byteContent = new byte[12];
+							buffer.read(byteContent);
+							assertEquals("Lorem\nIpsum\n", new String(byteContent));
+						}
+						else if (part.getName().equals("barPart")) {
+							assertEquals("barPart", part.getName());
+							Optional<String> filename = part.getFilename();
+							assertFalse(filename.isPresent());
+							assertEquals("bar", part.getValue().block());
+						}
+						else {
+							fail();
+						}
+					})
```

Comment:
Could `exchange.getParts()` with `collectToMap` be used to get all parts first and then invoke a method that asserts all parts, which can also be called from `assertGetAllParts`?